cli/cmd/profile.go

```diff
@@ -94,25 +100,33 @@ func newCmdProfile() *cobra.Command {
 		Args: cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			options.name = args[0]
+			clusterDomain := defaultClusterDomain
+			var k8sAPI *k8s.KubernetesAPI
 
 			err := options.validate()
 			if err != nil {
 				return err
 			}
-
-			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
-			if err != nil {
-				return err
-			}
-
-			_, values, err := healthcheck.FetchCurrentConfiguration(cmd.Context(), k8sAPI, controlPlaneNamespace)
-			if err != nil {
-				return err
-			}
-
-			clusterDomain := values.GetGlobal().ClusterDomain
-			if clusterDomain == "" {
-				clusterDomain = defaultClusterDomain
+			// performs an online profile generation and access-check to k8s cluster to extract
+			// clusterDomain from linkerd configuration
+			// profile generation based on tap data requires access to k8s cluster
+			if !options.ignoreCluster {
+				var err error
+				k8sAPI, err = k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+
+				if err != nil {
+					return err
+				}
+
+				_, values, err := healthcheck.FetchCurrentConfiguration(cmd.Context(), k8sAPI, controlPlaneNamespace)
+				if err != nil {
+					return err
+				}
+
+				clusterDomain = values.GetGlobal().ClusterDomain
+				if clusterDomain == "" {
+					clusterDomain = defaultClusterDomain
+				}
```

Comment:
So we already set `clusterDomain` to `defaultClusterDomain` above. Therefore, I think we should set it only if the cluster's domain is not empty:

```suggestion
				if cd := values.GetGlobal().ClusterDomain; cd != "" {
					clusterDomain = cd
				}
```

Reply:
That's right :smile_cat: 
Working on this right now!